mbed_flasher/flash.py

```diff
@@ -29,79 +29,119 @@
 class Flash(object):
     """ Flash object, which manage flashing single device
     """
-    FLASHERS = []
-    SUPPORTED_TARGETS = {}
+    _flashers = []
+    supported_targets = {}
 
     def __init__(self):
         logger = Logger('mbed-flasher')
         self.logger = logger.logger
-        self.FLASHERS = self.__get_flashers()
-        self.SUPPORTED_TARGETS = self.__update_supported_targets()
+        self._flashers = self.__get_flashers()
+        self.supported_targets = self.__update_supported_targets()
 
     def get_supported_targets(self):
-        return self.SUPPORTED_TARGETS
+        """
+        :return: supported targets
+        """
+        return self.supported_targets
 
     def get_supported_flashers(self):
+        """
+        :return: supported flashers
+        """
         available_flashers = []
-        for Flasher in self.FLASHERS:
-            available_flashers.append(Flasher.name)
+        for flasher in self._flashers:
+            available_flashers.append(flasher.name)
         return available_flashers
 
     def __update_supported_targets(self):
+        """
+        :return: list of all supported targets
+        """
         all_supported_targets = []
-        for Flasher in self.FLASHERS:
-            supported_targets = Flasher.get_supported_targets()
+        for flasher in self._flashers:
+            supported_targets = flasher.get_supported_targets()
             all_supported_targets.extend(supported_targets)
         return all_supported_targets
 
     @staticmethod
     def __get_flashers():
+        """
+        :return: list of available flashers
+        """
         from mbed_flasher.flashers import AvailableFlashers
         return AvailableFlashers
 
     @staticmethod
     def get_flasher(flasher=None):
-        from flashers import AvailableFlashers
-        for Flasher in AvailableFlashers:
-            if Flasher.name.lower() == flasher.lower():
-                return Flasher
-        else:
-            return EXIT_CODE_REQUESTED_FLASHER_DOES_NOT_EXIST
+        """
+        :param flasher: None, if not given a flasher
+        :return: return available flasher if found, otherwise return exit code
+        """
+        from mbed_flasher.flashers import AvailableFlashers
```

Comment:
not really, the logic here is going to be changed soon, maybe I should add todo.